### PR TITLE
Makes cultist coats hold respective items, can actually make them and adds cosmetic versions to the loadout

### PR DESCRIPTION
**Pull request by Seris02**
## About The Pull Request

Adds cosmetic versions of narsian and ratvarian coats to the loadout (brass and runed coats).
Narsian winter coat can be made with twisted construction on a winter coat.
Ratvarian winter coat can be made with replica fabricator on a winter coat.
Both can hold their respective cult items.

Also lets assistant's formal winter coat be able to hold insuls.

## Why It's Good For The Game

Because more winter coats are nice, and makes it so these aren't just adminspawn only.

## Changelog
:cl:
add: added the runed and brass winter coats (cosmetic ratvarian/narsian)
tweak: how the narsian/ratvarian coats can be made
/:cl:

### CODE REVIEW

File: code/modules/clothing/suits/miscellaneous.dm [hunk 1]
```diff
@@ -762,23 +763,70 @@
...
+		user.dropItemToGround(src)
+		to_chat(user,"<span class='large_brass'>\"Amusing that you think you are fit to wear this.\"</span>")
+		to_chat(user,"<span class='userdanger'>Your skin burns where the coat touched your skin!</span>")
+		user.adjustFireLoss(rand(10,16))
+
 /obj/item/clothing/suit/hooded/wintercoat/narsie
 	name = "narsian winter coat"
 	icon_state = "coatnarsie"
 	item_state = "coatnarsie"
+	armor = list("melee" = 30, "bullet" = 20, "laser" = 30,"energy" = 10, "bomb" = 30, "bio" = 10, "rad" = 10, "fire" = 30, "acid" = 30)
```
Comment (Trilbyspaceclone) [inline code/modules/clothing/suits/miscellaneous.dm:793]: armor = list("melee" = 30, "bullet" = 20, "laser" = 0,"energy" = 10, "bomb" = 30, "bio" = 10, "rad" = 10, "fire" = 30, "acid" = 30)
So its the same as Clock work coats.

**Review by Trilbyspaceclone — CHANGES_REQUESTED** [review]

File: code/modules/clothing/suits/miscellaneous.dm (hunk 1 above)
Comment (ghost) [reply code/modules/clothing/suits/miscellaneous.dm:793]: cwc armor makes you take extra laser damage

Comment (Trilbyspaceclone) [reply code/modules/clothing/suits/miscellaneous.dm:793]: Nar armor as always been also weaker to laser then bullet/melee

Comment (Seris02) [reply code/modules/clothing/suits/miscellaneous.dm:793]: funny thing is I took that directly from empowered armour and scaled it down, it's actually the ratvar one which isn't correct.

Comment (Seris02) [reply code/modules/clothing/suits/miscellaneous.dm:793]: which is now fixed.

File: code/modules/clothing/suits/miscellaneous.dm [hunk 2]
```diff
@@ -762,23 +763,70 @@
...
+		user.adjustBruteLoss(rand(10,16))
 
 /obj/item/clothing/head/hooded/winterhood/narsie
 	icon_state = "winterhood_narsie"
 
+/obj/item/clothing/suit/hooded/wintercoat/ratvar/fake
+	name = "brass winter coat"
+	icon_state = "coatratvar"
+	item_state = "coatratvar"
+	allowed = list(/obj/item/flashlight, /obj/item/tank/internals/emergency_oxygen, /obj/item/tank/internals/plasmaman, /obj/item/toy, /obj/item/storage/fancy/cigarettes, /obj/item/lighter)
```
Comment (Poojawa) [inline code/modules/clothing/suits/miscellaneous.dm]: you don't need to worry about this if the original one holds the same, it'll inherit just fine.

File: modular_citadel/code/modules/client/loadout/suit.dm [hunk 3]
```diff
@@ -69,6 +69,16 @@
...
+/datum/gear/coat/runed
+	name = "Runed winter coat"
+	category = SLOT_WEAR_SUIT
+	path = /obj/item/clothing/suit/hooded/wintercoat/narsie/fake
+
+/datum/gear/coat/brass
+	name = "Brass winter coat"
+	category = SLOT_WEAR_SUIT
+	path = /obj/item/clothing/suit/hooded/wintercoat/ratvar/fake
+
```
Comment (Ghommie) [inline modular_citadel/code/modules/client/loadout/suit.dm:81]: I'm fairly against loadouts containing items that could be easily mistaken for antag gear.
But I'm not against the arcade prizes pool containing such trickeries. :clown_face:

File: code/modules/antagonists/cult/blood_magic.dm [hunk 4]
```diff
@@ -585,6 +585,13 @@
 				new /obj/item/stack/sheet/runed_metal(T,quantity)
 				to_chat(user, "<span class='warning'>A dark cloud emanates from you hand and swirls around the plasteel, transforming it into runed metal!</span>")
 				SEND_SOUND(user, sound('sound/effects/magic.ogg',0,1,25))
+		if(istype(target, /obj/item/clothing/suit/hooded/wintercoat))
+			var/obj/item/clothing/suit/hooded/wintercoat/candidate = target
+			if (do_after(user,30,target=candidate))
+				new /obj/item/clothing/suit/hooded/wintercoat/narsie(T)
+				qdel(candidate)
+				to_chat(user, "<span class='warning'>A dark cloud emanates from you hand and swirls around the winter coat, transforming it into a narsian winter coat!</span>")
```
Comment (Ghommie) [inline code/modules/antagonists/cult/blood_magic.dm]:
```suggestion
				to_chat(user, "<span class='warning'>A dark cloud emanates from you hand and swirls around [target], transforming it into a narsian winter coat!</span>")
```

File: code/modules/antagonists/cult/blood_magic.dm [hunk 5]
```diff
@@ -585,6 +585,13 @@
 				new /obj/item/stack/sheet/runed_metal(T,quantity)
 				to_chat(user, "<span class='warning'>A dark cloud emanates from you hand and swirls around the plasteel, transforming it into runed metal!</span>")
 				SEND_SOUND(user, sound('sound/effects/magic.ogg',0,1,25))
+		if(istype(target, /obj/item/clothing/suit/hooded/wintercoat))
```
Comment (Ghommie) [inline code/modules/antagonists/cult/blood_magic.dm]:
```suggestion
		if(istype(target, /obj/item/clothing/suit/hooded/wintercoat) && target.type != /obj/item/clothing/suit/hooded/wintercoat/narsie)
```
Imagine transforming real narsian winter coat into itself.

File: code/modules/antagonists/cult/blood_magic.dm [hunk 6]
```diff
@@ -585,6 +585,13 @@
...
+			var/obj/item/clothing/suit/hooded/wintercoat/candidate = target
+			if (do_after(user,30,target=candidate))
+				new /obj/item/clothing/suit/hooded/wintercoat/narsie(T)
+				qdel(candidate)
```
Comment (Ghommie) [inline code/modules/antagonists/cult/blood_magic.dm]: Just use `target` instead of defining a new variable `candidate`.

**Review by Ghommie — CHANGES_REQUESTED** [review]
This will be a rather unknown feature for certain, but it's something.

File: modular_citadel/code/modules/client/loadout/suit.dm (hunk 3 above)
Comment (Seris02) [reply modular_citadel/code/modules/client/loadout/suit.dm:81]: seems fair, I'll do that

Comment (Seris02) [reply modular_citadel/code/modules/client/loadout/suit.dm:81]: actually nah, it'll just be like the tactitool turtleneck, if that's fine?

Comment (Ghommie) [reply modular_citadel/code/modules/client/loadout/suit.dm:81]: Ok fine. :/

**Review by Ghommie — APPROVED** [review]

**Pull request merged**